compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -1272,7 +1272,7 @@ public void visit(BLangSimpleVarRef varRefExpr) {
             BLangArrowFunction bLangArrowFunction = (BLangArrowFunction) enclEnv.node;
 
             for (BLangSimpleVariable param : bLangArrowFunction.params) {
-                if (param.symbol == symbol) {
+                if (param.symbol == getOriginalSymbol(symbol)) {
```

Comment:
Wouldn't it be better to use the original symbol when defining the symbol variable, instead of changing it here? (In L1175)

Reply:
We are not changing anything here, just comparing to see if it corresponds to a parameter. I don't think changing L1175 is correct, since `symbol` refers to the symbol for this particular variable reference expression.